trinity/extensibility/plugin_manager.py

```diff
@@ -74,7 +107,53 @@ def broadcast(self, event: BaseEvent, exclude: BasePlugin = None) -> None:
             if not plugin.should_start():
                 continue
 
-            plugin.start(None)
+            context = self._create_context_for_plugin(plugin)
+            plugin.start(context)
```

Comment:
I'm thinking that we should be really intentional/careful about this lifecycle.  Something about starting the next plugin mid-loop stands out to me as potentially problematic.

Consider this.

- `A` broadcasts event `x`
- Event `x` triggers `B` to start, broadcasting the `started` event.
- The `started` event triggers `C` to start.
- Finally `x` resumes being proadcast to the remaining plugins.

I see you have a comment about `start` being a temporary API so maybe this is a non issue.  Just wanted to surface it.

Reply:
I agree. I think that whole thing will change.